build/build.c

```diff
@@ -274,17 +373,22 @@ static rpmRC buildSpec(BTA_t buildArgs, rpmSpec spec, int what)
 exit:
     free(cookie);
     spec->rootDir = NULL;
-    if (rc != RPMRC_OK && rpmlogGetNrecs() > 0) {
+    if (rc != RPMRC_OK && rc != RPMRC_MISSINGBUILDREQUIRES &&
+	    rpmlogGetNrecs() > 0) {
 	rpmlog(RPMLOG_NOTICE, _("\n\nRPM build errors:\n"));
 	rpmlogPrint(NULL);
     }
     rpmugFree();
-
+    if (missing_buildreqs && !rc) {
+	rc = RPMRC_MISSINGBUILDREQUIRES;
+    }
+    if (rc == RPMRC_FAIL)
+	rc = 1;
```

Comment:
Please use consistent style with the to if's: either curly braces in both, or neither.